src/Google.Cloud.Functions.Invoker/FunctionsExtensions.cs

```diff
@@ -0,0 +1,159 @@
+        /// <returns>The original builder, for method chaining.</returns>
+        public static ILoggingBuilder AddFunctionsConsoleLogging(this ILoggingBuilder builder, WebHostBuilderContext context)
+        {
+            var options = FunctionsFrameworkOptions.FromConfiguration(context.Configuration);
+            ILoggerProvider provider = options.JsonLogging
+                ? new FactoryLoggerProvider(category => new JsonConsoleLogger(category))
+                : new FactoryLoggerProvider(category => new SimpleConsoleLogger(category));
+            builder.AddProvider(provider);
+            return builder;
+        }
+
+        /// <summary>
+        /// Configures Kestrel to listen to the port and address specified in the Functions Framework configuration.
+        /// </summary>
+        /// <param name="builder">The web host builder to configure.</param>
+        /// <returns>The original builder, for method chaining.</returns>
+        public static IWebHostBuilder ConfigureKestrelForFunctionsFramework(this IWebHostBuilder builder) =>
+            builder.ConfigureKestrel((context, serverOptions) =>
+            {
+                var options = FunctionsFrameworkOptions.FromConfiguration(context.Configuration);
+                serverOptions.Listen(options.GetIPAddress(), options.Port);
+            });
+
+        /// <summary>
+        /// Configures the services within the given web host builder for the Functions Framework to use
+        /// a target function from the given assembly. If the Functions Framework configuration (typically provided by
+        /// command line arguments or environment variables) does not specify a target function, the assembly is
+        /// scanned for a single compatible function type.
+        /// </summary>
+        /// <param name="builder">The web host builder to configure.</param>
+        /// <param name="assembly">The assembly expected to contain the Functions Framework target function.</param>
+        /// <returns>The original builder, for method chaining.</returns>
+        public static IWebHostBuilder AddFunctionTarget(this IWebHostBuilder builder, Assembly assembly) =>
+            builder.ConfigureServices((context, serviceCollection) =>
+                HostingInternals.AddServicesForFunctionTarget(serviceCollection, HostingInternals.GetFunctionTarget(context, assembly)));
+
+        /// <summary>
+        /// Configures the services within the given web host builder for the Functions Framework to use
+        /// the function target type specified by the <typeparamref name="TFunction"/> type parameter.
+        /// </summary>
+        /// <typeparam name="TFunction">The function target type.</typeparam>
+        /// <param name="builder">The web host builder to configure.</param>
+        /// <returns>The original builder, for method chaining.</returns>
+        public static IWebHostBuilder AddFunctionTarget<TFunction>(this IWebHostBuilder builder) =>
```

Comment:
I've been thinking about this a bit more, and it probably does make more sense for this to be on IServiceCollection. Let me know if you agree, and I'll amend the PR.

Reply:
(This is now done.)